protected/widgets/views/patientDiagnosesAndMedicationsWidget.php

```diff
@@ -32,20 +32,11 @@
                             </tr>
                             </thead>
                             <tbody>
-                            <?php foreach ($this->patient->episodes as $episode):
-                                if ($episode->diagnosis !== null): ?>
+                            <?php foreach ($this->patient->diagnoses as $diagnosis):?>
                                 <tr>
-                                    <td><?php echo CHtml::encode($episode->diagnosis->fully_specified_name) . ' (Principal)'; ?></td>
-                                    <td><?php echo CHtml::encode($episode->firm->name); ?></td>
-                                    <td><?php echo $episode->NHSDate('start_date'); ?></td>
-                                </tr>
-                            <?php endif;
-                            endforeach; ?>
-                            <?php foreach ($this->patient->secondarydiagnoses as $diagnosis): ?>
-                                <tr>
-                                    <td><?php echo CHtml::encode($diagnosis->disorder->fully_specified_name) . ' (Secondary)'; ?></td>
-                                    <td>N/A</td>
-                                    <td><?php echo CHtml::encode($diagnosis->dateText); ?></td>
+                                    <td><?php echo CHtml::encode($diagnosis) . ' (' . ($diagnosis->principal == 1 ? 'Principal' : 'Secondary') . ')'; ?></td>
+                                    <td><?php echo CHtml::encode($diagnosis->element_diagnoses->event->episode->firm->getNameAndSubspecialty()); ?></td>
```

Comment:
This is a fairly long chain of calls, can we guarantee that all of these will exist?

Reply:
Diagnosis will not actually exist if the events & episodes do not.